crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -296,11 +296,23 @@ impl<'a> Checker<'a> {
     pub(crate) fn generator(&self) -> Generator {
         Generator::new(
             self.stylist.indentation(),
-            self.f_string_quote_style().unwrap_or(self.stylist.quote()),
+            self.preferred_quote(),
             self.stylist.line_ending(),
         )
     }
 
+    /// Return the preferred quote for a generated `StringLiteral` node, given where we are in the
+    /// AST.
+    pub(crate) fn preferred_quote(&self) -> Quote {
+        self.f_string_quote_style().unwrap_or(self.stylist.quote())
+    }
+
+    /// Return the default string flags a generated `StringLiteral` node should use, given where we
+    /// are in the AST.
+    pub(crate) fn default_string_flags(&self) -> ast::StringLiteralFlags {
```

Comment:
It looks like this is now only actually used within this module; I think we can make it private!

```suggestion
    fn default_string_flags(&self) -> ast::StringLiteralFlags {
```

Reply:
This is still used by a bunch of rules on my branch. Did I miss a patch somewhere?

Reply:
Oops, sorry, I meant to suggest making `Checker::preferred_quote()` private, not `Checker::default_string_flags()`! _That_ one is now only used within this module 😄

Reply:
Oh of course, thanks for clarifying!